portability-gateway/src/main/java/org/dataportabilityproject/gateway/reference/DataTransferHandler.java

```diff
@@ -0,0 +1,153 @@
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.dataportabilityproject.gateway.reference;
+
+import static com.google.common.net.HttpHeaders.CONTENT_TYPE;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Preconditions;
+import com.google.common.net.HttpHeaders;
+import com.google.inject.Inject;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import java.io.IOException;
+import java.net.HttpCookie;
+import java.nio.charset.StandardCharsets;
+import org.dataportabilityproject.gateway.ApiSettings;
+import org.dataportabilityproject.gateway.action.createjob.CreateJobAction;
+import org.dataportabilityproject.gateway.action.createjob.CreateJobActionRequest;
+import org.dataportabilityproject.gateway.action.createjob.CreateJobActionResponse;
+import org.dataportabilityproject.gateway.reference.ReferenceApiUtils.HttpMethods;
+import org.dataportabilityproject.spi.cloud.storage.JobStore;
+import org.dataportabilityproject.spi.cloud.types.JobAuthorization;
+import org.dataportabilityproject.spi.cloud.types.PortabilityJob;
+import org.dataportabilityproject.spi.cloud.types.TypeManager;
+import org.dataportabilityproject.spi.gateway.auth.AuthDataGenerator;
+import org.dataportabilityproject.spi.gateway.auth.AuthServiceProvider;
+import org.dataportabilityproject.spi.gateway.auth.AuthServiceProviderRegistry;
+import org.dataportabilityproject.spi.gateway.types.AuthFlowConfiguration;
+import org.dataportabilityproject.types.client.transfer.DataTransferRequest;
+import org.dataportabilityproject.types.client.transfer.DataTransferResponse;
+import org.dataportabilityproject.types.client.transfer.DataTransferResponse.Status;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * HttpHandler for the {@link CreateJobAction}.
+ */
+final class DataTransferHandler implements HttpHandler {
+
+  private static final Logger logger = LoggerFactory.getLogger(DataTransferHandler.class);
+  public static final String PATH = "/_/DataTransfer";
+  public static final String ERROR_PATH = "/error";
+  private final CreateJobAction createJobAction;
+  private final AuthServiceProviderRegistry registry;
+  private final ApiSettings apiSettings;
+  private final JobStore store;
+  private final ObjectMapper objectMapper;
+
+  @Inject
+  DataTransferHandler(CreateJobAction createJobAction, AuthServiceProviderRegistry registry,
```

Comment:
the createJobAction is something that will be initialized via dep injection right? In which module would that be initially created?

Reply:
Good point, added modules to each package.